libexec/rc/rc.subr

```diff
@@ -2061,6 +2061,24 @@ find_local_scripts_old() {
 	done
 }
 
+_add_local_script()
+{
+	local file preferred service
+	file="$1"
+	service=${file##*/}
+	if [ -x "/etc/rc.d/${service}" ]; then
+		preferred="/etc/rc.d/${service}"
+	else
+		eval preferred=\$_local_script_${service}
+	fi
+	if [ -n "$preferred" -a "$preferred" != "$file" ]; then
+		warn "${service}: ignoring $file in favor of $preferred"
+		continue
+	fi
+	local_rc="${local_rc} ${file}"
+	eval _local_script_${service}="$file"
```

Comment:
Can `$file` include `-`? If so, `_local_script_${service}` couldn't be defined as `-` is disallowed in shell variable names.

I can see at zeno:
```
[kw543@zeno ~]$ ls -l /usr/local/etc/rc.d/*-*
-r-xr-xr-x  1 root  wheel   945 Jan 14  2021 /usr/local/etc/rc.d/avahi-daemon
-r-xr-xr-x  1 root  wheel  1197 Jan 14  2021 /usr/local/etc/rc.d/avahi-dnsconfd
-rwxr-xr-x  1 root  wheel  1294 Oct 10  2020 /usr/local/etc/rc.d/php-fpm
[kw543@zeno ~]$ 
```

Reply:
We'd probably have to replace `-` with `__` not to conflict with files having the same name but `_` instead of `-`.

Reply:
I've partially handled this case. One could do something fancier by replacing `ltr` use with a more complete escaper, but this should be good enough.